guides/source/action_view_overview.md

```diff
@@ -443,11 +383,31 @@ You can also specify a second partial to be rendered between instances of the ma
 <%= render partial: @products, spacer_template: "product_ruler" %>
 ```
 
-Rails will render the `_product_ruler` partial (with no data passed to it) between each pair of `_product` partials.
+Rails will render the `_product_ruler.html.erb` partial (with no data passed to it) between each pair of `_product.html.erb` partials.
+
+### Counter Variables
+
+Rails also makes a counter variable available within a partial called by the collection. The variable is named after the title of the partial followed by `_counter`. For example, when rendering a collection `@products` the partial `_product.html.erb` can access the variable `product_counter`. The variable indexes the number of times the partial has been rendered within the enclosing view, starting with a value of `0` on the first render.
+
+```erb
+# index.html.erb
+<%= render partial: "product", collection: @products %>
+```
+
+```erb
+# _product.html.erb
+<%= product_counter %> # 0 for the first product, 1 for the second product...
+```
```

Comment:
Since these ERB, we might have to comment using the tags too for proper highlighting:

<details><summary>Details</summary>
<p>

_before_
<img width="643" alt="Screenshot 2024-03-27 at 11 23 46" src="https://github.com/bhumi1102/rails/assets/26328/c23d151b-2db8-4a66-ab05-d1dc58f3b5ae">

_after_
<img width="657" alt="Screenshot 2024-03-27 at 11 26 31" src="https://github.com/bhumi1102/rails/assets/26328/fbb3868f-25a6-40e2-98a5-b9d44d33ca21">

</p>
</details> 


````suggestion
```erb
<%# index.html.erb %>
<%= render partial: "product", collection: @products %>
```

```erb
<%# _product.html.erb %>
<%= product_counter %> # 0 for the first product, 1 for the second product...
```
````